bin/read_visium_mtx.py

```diff
@@ -0,0 +1,157 @@
+        library_id = 'library_id'
+
+    adata.uns["spatial"][library_id] = dict()
+
+    if load_images:
+        files = dict(
+            tissue_positions_file=path / 'spatial/tissue_positions_list.csv',
+            scalefactors_json_file=path / 'spatial/scalefactors_json.json',
+            hires_image=path / 'spatial/tissue_hires_image.png',
+            lowres_image=path / 'spatial/tissue_lowres_image.png',
+        )
+
+        # check if files exists, continue if images are missing
+        for f in files.values():
+            if not f.exists():
+                if any(x in str(f) for x in ["hires_image", "lowres_image"]):
+                    print("You seem to be missing an image file.")
+                    print("Could not find '{f}'.")
+                    #logg.warning(
+                    #    f"You seem to be missing an image file.\n"
+                    #    f"Could not find '{f}'."
+                    #)
+                else:
+                    raise OSError(f"Could not find '{f}'")
+
+        adata.uns["spatial"][library_id]['images'] = dict()
+        for res in ['hires', 'lowres']:
+            try:
+                adata.uns["spatial"][library_id]['images'][res] = imread(
+                    str(files[f'{res}_image'])
+                )
+            except Exception:
+                raise OSError(f"Could not find '{res}_image'")
+
+        # read json scalefactors
+        adata.uns["spatial"][library_id]['scalefactors'] = json.loads(
+            files['scalefactors_json_file'].read_bytes()
+        )
+
+        #adata.uns["spatial"][library_id]["metadata"] = {
+        #    k: (str(attrs[k], "utf-8") if isinstance(attrs[k], bytes) else attrs[k])
+        #    for k in ("chemistry_description", "software_version")
+        #    if k in attrs
+        #}
+        
+        adata.uns["spatial"][library_id]["metadata"] = {k: "NA" for k in ("chemistry_description", "software_version")}
+
+        # read coordinates
+        positions = pd.read_csv(files['tissue_positions_file'], header=None)
+        positions.columns = [
+            'barcode',
+            'in_tissue',
+            'array_row',
+            'array_col',
+            'pxl_col_in_fullres',
+            'pxl_row_in_fullres',
+        ]
+        positions.index = positions['barcode']
+
+        adata.obs = adata.obs.join(positions, how="left")
+
+        adata.obsm['spatial'] = adata.obs[
+            ['pxl_row_in_fullres', 'pxl_col_in_fullres']
+        ].to_numpy()
+        adata.obs.drop(
+            columns=['barcode', 'pxl_row_in_fullres', 'pxl_col_in_fullres'],
+            inplace=True,
+        )
+
+        # put image path in uns
+        if source_image_path is not None:
+            # get an absolute path
+            source_image_path = str(Path(source_image_path).resolve())
+            adata.uns["spatial"][library_id]["metadata"]["source_image_path"] = str(
+                source_image_path
+            )
+
+    return adata
```

Comment:
Is this file actually used? It looks like its import statement is commented out in the `script_read_st_data.py`, which contains another function with the same name.

Reply:
I had hard time importing function from the local file, so for now just copied it into the script